apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -53,7 +56,25 @@ const fetchPoolAveragesForDateRange = withCache(
     );
 
     if (res.poolSnapshots.length === 0) {
-      return [0, 0, "", []];
+      // Following Fabio's recomendation on #BAL-872
+      if (calculateDaysBetween(from, to) != 1) {
+        return [0, 0, "", []];
+      }
+      console.warn(
+        "No return on poolSnapshots, trying to fetch in a few days before",
+      );
+      const retyGQL = await pools.gql(network).poolSnapshotInRange({
+        poolId,
+        from: from - SECONDS_IN_DAY * 7,
+        to,
+      });
+
+      if (retyGQL.poolSnapshots.length === 0) {
+        return [0, 0, "", []];
+      }
+      res.poolSnapshots = retyGQL.poolSnapshots
```

Comment:
this get the last data and return. e.g. Monday TVL = 10, Tuesday nothing happened then TVL = 10 too